x-pack/plugins/threat_intelligence/common/types/indicator.ts

```diff
@@ -54,3 +69,12 @@ export const generateMockFileIndicator = (): Indicator => {
 
   return indicator;
 };
+
+/**
+ * Mapping between the indicator type and the {@link RawIndicatorFieldId}.
+ */
+export const indicatorTypeToFieldMap: { [id: string]: RawIndicatorFieldId } = {
+  file: RawIndicatorFieldId.FileSha256,
+  url: RawIndicatorFieldId.UrlFull,
+  ip: RawIndicatorFieldId.Ip,
```

Comment:
I know but couldn't find a better way to do this. I stole the logic from [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/threat_intelligence/public/modules/indicators/lib/display_value.ts#L32) where we return the actual value to display. We have a ruleArray with only 3 options: `isIpIndicator`, `isUrlIndicator` and `isFileIndicator` with the extract method hardcoded. To your knowledge do we have a better solution to do this?

Reply:
my main point that the logic in `add_to_timeline.tsx` 
```
field = indicatorTypeToFieldMap[unwrapValue(data, RawIndicatorFieldId.Type) || ''];
```
won't even work, because `unwrapValue(data, RawIndicatorFieldId.Type)` will return `ipv4-addr` and you don't have a mapping for that. Ofc the simplest would be to add all the possible keys from https://docs.google.com/spreadsheets/d/1oYx9CQodKVmr0OtlJcjm1mNoUBlDcbHai0UYxtP8YzY/edit#gid=1814426429 
```
file
url
email-addr
email (doesn't exist in STIX 2.0)
email-message
domain-name
domain
ipv4-addr
ipv6-addr
x509-certificate
x509 Serial (doesn't exist in STIX 2.0)
windows-registry-key
autonomous-system
suppress (doesn't exist in STIX 2.0)
mac-addr
unknown
```
to the mapping, but then when a new type is added I'm 99% confident that we won't add it to the mapping until a user complain :)
Let me think on how we could awoid that

Reply:
ok thanks for the pointer, I will spend time thinking about a good way to handle this!

Reply:
for now still with a very basic key-value object, like discussed a cleaner work will be done in [this ticket](https://github.com/elastic/security-team/issues/4720)